lib/Hashids.pm

```diff
@@ -221,7 +232,7 @@ sub _decode {
         push @$res => $self->_unhash( $part, \@alphabet );
     }
 
-    return unless $self->Hashids::encode(@$res) eq $orig;
+    return unless $self->encode(@$res) eq $orig;
```

Comment:
`Hashids::encode` is essentially `SUPER::encode` here, so it is needed for subclassing (which usually is just wrapping around `encode`.)